odoo/tests/common.py

```diff
@@ -455,6 +455,20 @@ def assertRaises(self, exception, func=None, *args, **kwargs):
         else:
             return self._assertRaises(exception, **kwargs)
 
+    @contextmanager
+    def assertNoLogs(self, logger: str, level: str):
+        if hasattr(unittest.TestCase, 'assertNoLogs'):  # Py3.10
+            return super().assertNoLogs(logger, level)
```

Comment:
What about doing the test once outside the method definition?
```suggestion
    if sys.version_info < (3, 10):
        # simplified backport of assertNoLogs()
        @contextmanager
        def assertNoLogs(self, logger: str, level: str):
            ...
```